torchmetrics/image/map.py

```diff
@@ -0,0 +1,369 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import sys
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
+
+import torch
+from torch import Tensor
+
+from torchmetrics.metric import Metric
+from torchmetrics.utilities.imports import _PYCOCOTOOLS_AVAILABLE
+
+if _PYCOCOTOOLS_AVAILABLE:
+    from pycocotools.coco import COCO
+    from pycocotools.cocoeval import COCOeval
+else:
+    COCO, COCOeval = None, None
+
+COCO_STATS_MAP_VALUE_INDEX = 0
```

Comment:
We're only returning MAP and MAR, but apparently the library computes a whole bunch of other values:

     Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ]
     Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ]
     Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ]
     Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ]
     Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ]
     Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ]
     Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ]
     Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ]
     Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ]
     Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ]
     Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ]
     Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ]

I would be interested in the other values too. Since they are computed anyway, hopefully we could return them too.

Reply:
It does. The reason I only added the two is that I assume they are the most relevant (usually only a single mAP value is given, as in [TF2](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md)).

Adding the others would require some boilerplate code. Do you think they are relevant or just nine to have?

Reply:
With the changes from https://github.com/PyTorchLightning/metrics/pull/467#discussion_r714596707 this gets easier now. Done

Reply:
For me the different AP values would be relevant. What makes them a little less usable is that the small/medium/large categories are hardcoded, but I would still be interested in all the values. I guess there's not so much code needed now that we don't have a separate state for every output?